### PR TITLE
[SPARK-44223][CORE] Remove LevelDB support

### PR DESCRIPTION
**Pull request by Hisoka-X**
### What changes were proposed in this pull request?
This PR remove LevelDB support

### Why are the changes needed?
The leveldb project seems to be no longer maintained, and we can always replace it with rocksdb.

### Does this PR introduce _any_ user-facing change?
Yes, the LevelDB will not supported in 4.1.0

### How was this patch tested?
exist test.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 1]
```diff
@@ -726,12 +726,12 @@ package object config {
...
     ConfigBuilder(Constants.SHUFFLE_SERVICE_DB_BACKEND)
       .doc("Specifies a disk-based store used in shuffle service local db. " +
-        "LEVELDB or ROCKSDB.")
+        "Only supported ROCKSDB at now.")
       .version("3.4.0")
       .stringConf
       .transform(_.toUpperCase(Locale.ROOT))
       .checkValues(DBBackend.values.map(_.toString).toSet)
-      .createWithDefault(DBBackend.LEVELDB.name)
+      .createWithDefault(DBBackend.ROCKSDB.name)
```
Comment (LuciferYang) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala:734]: I'm not against this change, but I'm quite conflicted about it.

For shuffle db, we didn't make `ROCKSDB` the default in Spark 3.5.0, so I'm not sure if it's appropriate to just remove it in Spark 4.0. This is why I haven't submitted a pr to address this issue.

Let's wait for other people's opinions.

Comment (mridulm) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala:734]: Agree, let us change to rocksdb as default first.
We can always remove it in 4.1 or later

**Comment by LuciferYang** [comment]
@Hisoka-X Can you restart this job? I believe we can still remove Leveldb's support for HybridStore in Spark 4.0.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala (hunk 1 above)
Comment (Hisoka-X) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Sure, let me clean it up again.

Comment (Hisoka-X) [reply core/src/main/scala/org/apache/spark/internal/config/package.scala]: Hi @LuciferYang , I did some update to this PR. Could you help to reopen it?

**Pull request closed without merging**